ci/perf-test-cloudbuild.yaml

```diff
@@ -0,0 +1,105 @@
+---
+# Copyright 2023 Google LLC All Rights Reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+#
+# Google Cloud Builder CI configuration
+#
+steps:
+  #
+  # Creates the initial make + docker build platform
+  #
+  - name: ubuntu
+    args:
+      - bash
+      - -c
+      - "echo 'FROM gcr.io/cloud-builders/docker\nRUN apt-get install make\nENTRYPOINT\
+        \ [\"/usr/bin/make\"]' > Dockerfile.build"
+    waitFor: ['-']
+
+  - name: gcr.io/cloud-builders/docker
+    id: build-make-docker
+    args: [build, -f, Dockerfile.build, -t, make-docker, .]  # we need docker and make to run everything.
+
+  #
+  # pull the main build image if it exists
+  #
+  - name: make-docker
+    id: pull-build-image
+    dir: build
+    env: ['REGISTRY=${_REGISTRY}', 'CUSTOM_LOCAL_TAG=performance-test']
+    args: [pull-build-image-with-tag]
+
+  #
+  # Run the performance tests with default feature gates setting
+  #
+  - name: performance-test
+    script: |
```

Comment:
This "feels" long enough that we should maybe bump it out of YAML and into its own thing. WDYT?